src/components/form/PasswordInputField.tsx

```diff
@@ -0,0 +1,58 @@
+import IconButton from "@mui/material/IconButton";
+import VisibilityIcon from "@mui/icons-material/Visibility";
+import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
+import { useTheme } from "@mui/material";
+import { FunctionComponent, useState } from "react";
+import { TextInputField } from "components";
+
+interface PasswordInputFieldProps {
+  readonly ariaLabel?: string;
+  readonly endAdornmentHandler?: () => void;
+  readonly externalMaskPassword?: boolean;
+  readonly name: string;
+  readonly placeholder?: string;
+  readonly showEndAdornment?: boolean;
+}
+
+const PasswordInputField: FunctionComponent<PasswordInputFieldProps> = ({
+  ariaLabel = "Password input field",
+  endAdornmentHandler,
+  externalMaskPassword = true,
+  name,
+  placeholder = "Password",
+  showEndAdornment = true,
+}) => {
+  const theme = useTheme();
+  const [internalMaskPassword, setInternalMaskPassword] = useState(true);
+
+  const isMasked = endAdornmentHandler ? externalMaskPassword : internalMaskPassword;
```

Comment:
Might be more readable to use 

```
const isMasked = externalMaskPassword ?? internalMaskPassword
```

The nullish coalescing operator will only reference the second variable if the first variable is `null` or `undefined`

Reply:
This is part of the confusing stuff i mentioned to you...on this line: `const isMasked = endAdornmentHandler ? externalMaskPassword : internalMaskPassword;`  I am checking if there is a handler passed through... if so, we want to use the `externalMaskPassword` prop if none is provided, I am creating an internal masked state and toggling that value

Reply:
Yeah, that works. I was thinking it might simplify the code if we just check the `externalMaskPassword` prop and use that to determine whether to use the internal value or not. The user probably wouldn't pass an `externalMaskPassword` prop unless they were also handling the end adornment funtionality externally, so I think it would be the same in regards to functionality. Either way sounds good to me though.

Reply:
ohhh i see what you meant, the implementer of PasswordInputField would not pass externalMaskPassword without the handlePressEndAdornment... i tested it and it worked, i do agree it reads better that way. change coming in the next push